### ISSUE TITLE
Zipped ROMS causing crashes in WinXP

### ISSUE THREAD

**Issue by sa666666** [stella-emu/stella#101]
According to PM from a tester, this has been happening since Stella 4.1.  I am able to reproduce it, but not yet debug it, since I don't have a WinXP system set up to build Stella.

Also, it happens only intermittently (but consistently after 5-10 tries), which will obviously make it that much harder to track down.

**Comment by sa666666** [comment]
I'm having an extremely hard time creating a proper build environment for Windows XP.  I can't install a new enough Visual Studio for the current code to compile, so I can't even get a debugger installed to track what the problem is.  I have a hard decision to make for the 5.0 release; I may have to drop WinXP support entirely.

**Comment by thrust26** [comment]
XP is 16 year old now and support ended 3 years ago. It is dead. Period.

**Comment by DirtyHairy** [comment]
XP is dead, buried and decomposed :smile: I'd drop it. Besides, at least this issue has a simple workaround: don't use zipped cartridges.

**Comment by sa666666** [comment]
It the coming months it may no longer be an issue anyway.  Once I move to Visual Studio 2017, I suspect it won't support even building a binary for XP.  XP support was already supposed to be removed from VS 2015, but there was a backlash and support was added for XP/SP3 only.  I think this will disappear entirely in 2017.  So it might be better to break compatibility cleanly for 5.0.

**Comment by sa666666** [comment]
Well, it looks like WinXP gets a small reprieve after all.  I spent this afternoon first trying to locate Visual Studio 2010 through my Microsoft Academic Alliance account, and finally just got it from a torrent :see_no_evil:

Once I had it installed in Windows XP, I compiled both Stella 4.0 and 4.1.  And sure enough, the breakage is in 4.1, with basically a buffer overflow.  AFAICT, it only happens in Windows XP when using a zipped ROM of size less than 4K (this is why Combat was failing).  Why it didn't cause an issue on any other system is a mystery, since it technically should have crashed elsewhere too.

Anyway, never let it be said that I will let a bug go without a fight  :smile:  Issue fixed in https://github.com/stella-emu/stella/commit/a2c98e2409c12882823c055f849b8c3443ec9ddb.
